### DISCUSSION TITLE
How to get current test name?

### DISCUSSION THREAD

**Discussion by egoist** [vitest-dev/vitest#297] [Q&A]
In Jest you use `expect.getState().currentTestName`, not sure what's the API in Vitest 🤔

**Comment by antfu** [comment]
We could add an API for that. I am not a fan of Jest's approach here, not sure if we should align with it 🤔

**Reply by egoist** [reply]
I would like some API like this:

```js
test('simple', t => {
    t.title // simple
})
```

**Reply by sheremet-va** [reply]
We already have `getState` function, might as well put a name inside 👀

**Reply by jbergstroem** [reply]
`expect.getState().testPath` would be useful as well..

**Reply by sheremet-va** [reply]
This is now implemented 👀

**Reply by tompahoward** [reply]
Is this supposed to be available in `beforeEach`? For me `expect.getState().currentTestName` is undefined in `beforeEach` and I'm wondering if I've misconfigured something

**Reply by Any2suited66** [reply]
I'm seeing the same issue as @tompahoward.  Did you get it figured out?  I'm new to Jest and JS in general.

**Reply by sheremet-va** [reply]
It's not available in `beforeEach`, which is a bug. But you can still access it with context:

```ts
beforeEach({ expect, meta }) => {
  // expect.getState().currentTestName
  // meta.name
})
```

**Reply by DysphoricUnicorn** [reply]
> It's not available in `beforeEach`, which is a bug. But you can still access it with context:
> 
> ```ts
> beforeEach({ expect, meta }) => {
>   // expect.getState().currentTestName
>   // meta.name
> })
> ```


Is this still available somehow in v1?

I can't see it mentioned in the changelog but my tests broke after the update because of it

**Reply by hi-ogawa** [reply]
It would be easier to investigate if you can provide a reproduction. Please feel free to create an issue if something was working but now broken.

**Reply by sheremet-va** [reply]
The meta attribute is not available. Use `task` instead. It was deprecated for some time and removed in 1.0 I think.
